packages/app-page-builder/src/blockEditor/Editor.tsx

```diff
@@ -48,11 +48,8 @@ export const BlockEditor: React.FC = () => {
             .then(({ data }) => {
                 const blockCategoriesData: PbBlockCategory[] =
                     get(data, "pageBuilder.listBlockCategories.data") || [];
-                blockCategoriesData.forEach(element => {
-                    createBlockCategoryPlugin({
-                        ...element
-                    });
-                });
+
```

Comment:
To make this part reusable, I would move this whole section of `const blockCategories = client.query(...)` into the `useBlockCategories` hook, and then you also don't need to pass blocks to editor state initializer. So you would reuse that hook from both the block editor, as well as the page editor.

It just crossed my mind that you could also just use Apollo cache as state management for block categories, so you won't even need a recoil atom 🤦 . You basically just have the `useQuery` hook wrapped into a `useBlockCategories` hook, and all of a sudden it's reusable everywhere without any state management.

Something like this:
```ts
export const useBlockCategories() {
    const { data } = useQuery(LIST_BLOCK_CATEGORIES);

    return get(data, "pageBuilder.listBlockCategories.data") || [];
}
```

What do you think?